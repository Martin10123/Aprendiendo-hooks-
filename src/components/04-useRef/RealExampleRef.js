import React, { useState } from "react";
import { MultipleCustomHooks } from "../03-example/MultipleCustomHooks";

const RealExampleRef = () => {
  const [show, setshow] = useState(false);

  return (
    <div>
      <h1>Real example ref</h1>
      <hr />

      {show && <MultipleCustomHooks />}

      <button onClick={() => setshow(!show)} className="btn btn-primary mt-5">
        Show/hide
      </button>
    </div>
  );
};

export default RealExampleRef;
