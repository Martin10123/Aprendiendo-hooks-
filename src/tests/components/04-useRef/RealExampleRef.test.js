import React from "react";
// import { renderHook } from "@testing-library/react-hooks";

import { shallow } from "enzyme";
import RealExampleRef from "../../../components/04-useRef/RealExampleRef";

describe("Pruebas en el componente <RealExampleRef/>", () => {
  const wrapper = shallow(<RealExampleRef />);
  test("debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  test("debe de mostrar el componente <MultipleCustomHooks />", () => {
    wrapper.find("button").simulate("click");

    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
