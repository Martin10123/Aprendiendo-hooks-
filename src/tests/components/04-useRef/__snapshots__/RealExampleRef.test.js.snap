// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <RealExampleRef/> debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    Real example ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/hide
  </button>
</div>
`;
